infra/scripts/language/sk_clients.py

```diff
@@ -0,0 +1,410 @@
+config['clu_project_name'] = os.environ.get("CLU_PROJECT_NAME")
+config['clu_deployment_name'] = os.environ.get("CLU_DEPLOYMENT_NAME")
+config['cqa_project_name'] = os.environ.get("CQA_PROJECT_NAME")
+config['cqa_deployment_name'] = os.environ.get("CQA_DEPLOYMENT_NAME")
+
+# Create chat completion service using Azure OpenAI - not using right now but good to have as reference
+# chat_completion_service = AzureChatCompletion(
+#     endpoint=AOAI_ENDPOINT,
+#     deployment_name=MODEL_NAME,
+# )
+# print('successfully created chat completion service')
+
+def create_tools(config):
+    # Set up the auth details for the OpenAPI connection
+    auth = OpenApiManagedAuthDetails(security_scheme=OpenApiManagedSecurityScheme(audience="https://cognitiveservices.azure.com/"))
+
+    # Read in the OpenAPI spec from a file
+    with open("../../openapi_specs/clu.json", "r") as f:
+        clu_openapi_spec = json.loads(bind_parameters(f.read(), config))
+
+    clu_api_tool = OpenApiTool(
+        name="clu_api",
+        spec=clu_openapi_spec,
+        description= "An API to extract intent from a given message - you MUST use version \"2023-04-01\" as this is extremely critical",
+        auth=auth
+    )
+
+    # Read in the OpenAPI spec from a file
+    with open("../../openapi_specs/cqa.json", "r") as f:
+        cqa_openapi_spec = json.loads(bind_parameters(f.read(), config))
+
+    # Initialize an Agent OpenApi tool using the read in OpenAPI spec
+    cqa_api_tool = OpenApiTool(
+        name="cqa_api",
+        spec=cqa_openapi_spec,
+        description= "An API to get answer to questions related to business operation",
+        auth=auth
+    )
+
+    return clu_api_tool, cqa_api_tool
+
+# Create custom selection strategy for the agent groupchat by sublcassing the SequentialSelection Strategy
+class SelectionStrategy(SequentialSelectionStrategy):
+    async def select_agent(self, agents, history):
+        """
+        Select agent based on the current message and agent.
+        This method overrides the default selection strategy to use a custom logic.
+            - The triage agent is always selected after the user message.
+            - If the triage agent returns a CQA result, the chat is terminated and the CQA result is returned.
+            - If the triage agent returns a CLU result, the head support agent is selected.
+            - Based on the intent and entities passed to the head support agent, the appropriate custom agent (return, refund, order status) is selected
+            - The custom agent returns the result and the chat is terminated.
+            - If any agents fail, the chat falls back to a default response.
+        """
+        last = history[-1] if history else None
+
+        print("last message:", last)
+        print("last message name:", last.name if last else None)
+                
+        if not last or last.role == AuthorRole.USER or last is None:
```

Comment:
nit - you already checked for `not last` you don't need to check for `last is None` as well

```
>>>
>>> x = None
>>> if not x:
...     print("not x")
... else:
...     print("x")
...
not x
>>>
```